### ISSUE TITLE
NES mapper 34 issue

### ISSUE THREAD

**Issue by nsx0r** [sanni/cartreader#770]
Hello, trying to dump the NES cartridge Impossible Mission II though the Sanni does not detect the cartridge. Manually selecting the cartridge results in a 64KB dump where it should be 128K. Cartridge edge connector is clean and the cartridge has no damage. Successfully dump other NES cartridges. Any advice welcome, thank you for your time.

The PCB version is **NINA-002 REV. A**, identical to:
[https://nescartdb.com/profile/view/1241/impossible-mission-ii](url)

Log as follows.

```
OSCR HW5 V12.4

[+] NES/Famicom


Searching database
for 21BF845C or 64B86EED...

CRC not found in database
Using manual selection

Printing PRG at 32768
4c 31 80 4c 20 03 4c 31 80 4c 30 80 4c da 80 4c 
11 81 4c a1 ae 4c 7d 82 4c 5b ac 4c 53 80 4c 17 
b0 4c bc b1 4c 71 ce 4c 3d d2 4c 82 ce 4c 43 d2 
40 a2 0a bd 3f 80 9d 00 01 ca 10 f7 4c 00 01 a9 
00 8d fd 7f 4c 00 80 48 a9 0c 4c 00 03 48 a9 0f 
4c 00 03 86 cd 84 ce 18 85 08 0a 0a 18 65 08 65 
56 85 c4 a9 00 65 57 85 c5 a0 00 b1 c4 85 c8 c8 
b1 c4 18 65 ce 85 c9 c8 b1 c4 18 69 cb 85 c6 c8 
b1 c4 69 8c 85 c7 c8 a5 3d 10 04 b1 c4 85 3d a9 
00 85 ca a9 00 85 cb a5 cd 85 cc 20 cb 80 a4 ca 
b1 c6 c8 84 ca a6 cc a4 ce 20 38 b0 a5 16 30 04 
a4 cb 91 7b e6 cb e6 cc a5 cb c5 c8 d0 e0 e6 ce 
a5 ce c5 c9 d0 cd a9 ff 85 3d 60 a2 64 a0 00 84 
7b 86 7c a6 cd a4 ce 4c 90 ae 18 69 02 85 08 a9 
00 85 09 a5 08 0a 26 09 0a 26 09 18 65 08 85 08 
a5 09 69 00 85 09 18 a5 08 65 56 85 08 a5 09 65 
57 85 09 a0 00 b1 08 aa c8 b1 08 a8 84 09 86 08 
60 85 7a 86 79 a2 60 a0 00 84 7b 86 7c a0 00 20 
4b 81 29 ff 10 0e 29 7f aa 20 4b 81 20 42 81 ca 
d0 fa f0 eb f0 14 aa 20 4b 81 20 42 81 ca d0 f7 
f0 dd 91 7b e6 7b d0 02 e6 7c 60 b1 79 e6 79 d0 
02 e6 7a 60 88 c0 ff d0 01 ca 8a 18 65 7a 85 7a 
8a 18 65 7c 85 7c b1 79 91 7b 88 d0 f9 b1 79 91 
7b 88 c6 7a c6 7c ca e0 ff d0 eb 60 ff 00 80 40 
c0 20 a0 60 e0 10 90 50 d0 30 b0 70 f0 08 88 48 
c8 28 a8 68 e8 18 98 58 d8 38 b8 78 f8 04 84 44 
c4 24 a4 64 e4 14 94 54 d4 34 b4 74 f4 0c 8c 4c 
cc 2c ac 6c ec 1c 9c 5c dc 3c bc 7c fc 02 82 42 
c2 22 a2 62 e2 12 92 52 d2 32 b2 72 f2 0a 8a 4a 
ca 2a aa 6a ea 1a 9a 5a da 3a ba 7a fa 06 86 46 
c6 26 a6 66 e6 16 96 56 d6 36 b6 76 f6 0e 8e 4e 
ce 2e ae 6e ee 1e 9e 5e de 3e be 7e fe 01 81 41 

Printing PRG at 57344
60 10 c1 14 31 4e 24 e4 d8 75 24 c1 08 21 42 15 
22 4e 4d 87 43 04 10 86 08 12 42 40 15 4e 20 70 
4a 0e 90 ca 0e c0 cd 0e f0 d0 0f 20 d3 0f 50 d6 
14 70 07 1c 73 07 44 75 84 0c b0 ce 0d 10 d4 40 
c0 00 34 00 0e 00 03 c0 01 00 00 30 10 cd 04 23 
81 04 f0 42 40 10 d8 58 46 56 11 a5 85 6d 61 1c 
58 46 97 04 22 8f 74 e5 9d 14 14 90 06 00 06 0a 
00 33 d4 0d 13 04 64 c1 1f 01 82 c4 1c b4 47 2c 
c3 c3 40 f0 52 3c 76 8f 08 45 8b 41 63 00 74 e4 
1d 29 f7 41 00 93 85 44 61 80 68 60 c6 50 a4 94 
25 65 07 69 50 c7 6d 24 9b 48 20 4f 04 10 87 04 
30 48 3d 50 00 71 60 42 08 70 84 70 83 d6 74 01 
8b 01 11 c1 34 a0 85 05 22 82 14 15 4a 08 50 50 
40 b1 81 0d 76 06 0c 87 80 0c 13 ea 52 a5 ea 3c 
c5 0c 5c c0 05 3c b5 0b 5c b1 cc 28 c7 08 20 32 
48 0c 72 43 20 90 89 24 22 89 0c 62 83 1c a0 88 
28 12 4a 04 a2 82 2c a0 c5 2c 31 8b 08 72 c1 20 
b0 49 2c 12 8b 04 b2 c2 30 b0 c8 30 12 4c 04 83 
41 24 d0 47 38 22 0e 04 93 81 28 e0 85 28 42 94 
24 02 94 69 56 69 09 a4 c0 00 00 44 00 06 09 5c 
36 c9 11 f0 1e 28 42 41 40 92 1a 30 11 0f 05 53 
c7 11 71 4a 5c c3 db 04 07 43 61 d1 00 24 81 d2 
44 74 85 58 91 d6 24 c5 8d 41 64 12 58 75 16 3d 
a7 03 15 c7 41 08 11 02 0d c0 04 74 00 01 5c 10 
81 18 20 c1 24 f5 c0 00 52 40 00 12 00 08 91 4d 
00 00 05 3c 00 01 34 00 8f 15 20 00 00 37 40 00 
14 80 09 d0 d5 00 00 05 0c 05 89 04 50 53 1c 25 
90 14 d0 57 48 21 c1 01 91 c4 0c 63 02 54 22 50 
2d 03 b9 4f 91 87 2c b2 cf 2d 32 c9 30 d3 0b 29 
02 91 29 26 52 29 36 53 29 46 54 29 56 55 65 61 
8e 60 03 d8 01 06 00 45 80 12 60 04 d8 00 31 0c 
74 44 c7 3d 3a 42 08 10 85 00 56 4c 30 34 cc 09 
Select Mapping from List

NES CART READER

CURRENT SETTINGS

MAPPER:   34
PRG SIZE: 64K
CHR SIZE: 64K
RAM SIZE: 0K

Press Button...

[+] Read iNES Rom


Saving to NES/ROM/ImpossibleMissi/6/...
[*******************]
CRC32... 0142560B -> Not found

Press Button...
```

_Originally posted by @N30Avalon in https://github.com/sanni/cartreader/discussions/769_

**Comment by nsx0r** [comment]
should be fixed in [commit 12d2a6c](https://github.com/sanni/cartreader/commit/12d2a6cac408f0939433dd930be7848139afff19)

**Comment by nsx0r** [comment]
confirmed fixed
